forum/views/flags.py

```diff
@@ -0,0 +1,94 @@
+"""Forum Flag API Views."""
+
+from rest_framework import status
+from rest_framework.permissions import AllowAny
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from forum.models.contents import Contents
+from forum.models.users import Users
+from forum.serializers.content import ContentSerializer
+from forum.utils import flag_as_abuse, un_flag_as_abuse
+
+
+class CommentFlagAPIView(APIView):
+    """
+    API View for flagging/unflagging comments.
+
+    Handles PUT requests to flag or unflag a comment.
+    """
+    permission_classes = (AllowAny,)
+
+    def put(self, request, comment_id, action):
+        """
+        Flag or unflag a comment.
+
+        Parameters:
+        request (Request): The incoming request.
+        comment_id (str): The ID of the comment to flag/unflag.
+        action (str): The action to take (either "flag" or "unflag").
+
+        Returns:
+        Response: A response with the updated comment data.
+        """
+        request_data = request.data
+        user = Users().get(_id=request_data["user_id"])
```

Comment:
The fact that we are accessing `_id` here is a leaky abstraction. We don't want to know that we are accessing a mongodb model. So maybe `_id` should be a non-named argument of the `.get` method. I.e: `get(request_data["user_id"])`.

Reply:
Done